### DISCUSSION TITLE
Make Stop and Skip Button work for Hires fix

### DISCUSSION THREAD

**Discussion by bdulfer** [vladmandic/sdnext#1128] [Ideas]
I'm using Hires fix quite a lot, because for me it gives slightly better results.
Generating continuously or using long wildcard lists helps a lot with finding the things I like.

Stopping or Skipping during these processes stops the image processing, which really helps. But still the Hires function is performed.

Here are two changes to `modules/processing.py` that will extend the stop/skip to Hires as well.
1. Remove lines 577 and 578 to keep the Skipped state for later
2. Change line 809 from 
   `        if not self.enable_hr:` to `        if not self.enable_hr or state.interrupted or state.skipped:` 
   and add 
   `            state.skipped = False` 
   after that to skip the Hires processing

I don't know if this is the best way to do this, but at least it works.

**Comment by vladmandic** [comment]
i'll add it (with some refactoring)

**Reply by bdulfer** [reply]
Works like a charm! 

Many thanks!
